lib/reflectAll.js

```diff
@@ -38,7 +39,43 @@ import reflect from './reflect';
  *     // results[1].error = Error('bad stuff happened')
  *     // results[2].value = 'two'
  * });
+ *
+ * // an example using an object instead of an array
+ * let tasks = {
+ *     one: function(callback) {
+ *         setTimeout(function() {
+ *             callback(null, 'one');
+ *         }, 200);
+ *     },
+ *     two: function(callback) {
+ *         callback('two');
+ *     },
+ *     three: function(callback) {
+ *         setTimeout(function() {
+ *             callback(null, 'three');
+ *         }, 100);
+ *     }
+ * };
+ *
+ * async.parallel(async.reflectAll(tasks),
+ * // optional callback
+ * function(err, results) {
+ *     // values
+ *     // results.one.value = 'one'
+ *     // results.two.error = 'two'
+ *     // results.three.value = 'three'
+ * });
  */
 export default function reflectAll(tasks) {
-    return tasks.map(reflect);
+    var results;
+    if (isArray(tasks)) {
+        results = tasks.map(reflect);
+    } else {
+        var keys = Object.keys(tasks);
+        results = {};
+        keys.forEach(function(key) {
+            results[key] = reflect.call(this, tasks[key]);
```

Comment:
Pretty sure this line doesnt work with async.auto  when a function has a dependency